feature_layers/create-mobile-geodatabase/src/main/java/com/esri/samples/create_mobile_geodatabase/CreateMobileGeodatabaseController.java

```diff
@@ -0,0 +1,326 @@
+              addFeatureFuture.addDoneListener(() -> {
+                try {
+                  addFeatureFuture.get();
+                  // update the total feature count on screen if feature was added successfully
+                  label.setText("Number of features added: " + geodatabaseFeatureTable.getTotalFeatureCount());
+
+                } catch (InterruptedException | ExecutionException ex) {
+                  throw new RuntimeException(ex);
+                }
+              });
+            }
+          }
+        }
+        else {
+          // close table before adding a new feature to the map
+          handleTableWindowVisibility();
+        }
+      });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * create geodatabase and feature layer from geodatabase feature table descriptions
+   */
+  @FXML
+  private void createGeodatabase() {
+
+    // get the path for the geodatabase file
+    geodatabasePath = Paths.get(System.getProperty("user.dir") + "/LocationHistory.geodatabase");
+    try {
+      Files.deleteIfExists(geodatabasePath);
+    } catch (IOException ioException) {
+      ioException.printStackTrace();
+    }
+
+    // create geodatabase from the specified mobile geodatabase file path
+    var geodatabaseFuture = Geodatabase.createAsync(geodatabasePath.toString());
+    geodatabaseFuture.addDoneListener(() -> {
+      try {
+        // get the instance of the mobile geodatabase
+        geodatabase = geodatabaseFuture.get();
+
+        // create a table description to store features as map points and set non-required properties to false
+        var tableDescription = new TableDescription("LocationHistory", SpatialReferences.getWgs84(),
+          GeometryType.POINT);
+
+        // Set up the fields for the table. FieldType.OID is the primary key of the SQLite table.
+        var fieldDescriptionOID = new FieldDescription("oid", Field.Type.OID);
+        var fieldDescriptionText = new FieldDescription("collection_timestamp", Field.Type.TEXT);
+        tableDescription.getFieldDescriptions().addAll(List.of(fieldDescriptionOID, fieldDescriptionText));
+
+        // add a new table to the geodatabase feature table by creating one from the table description
+        var geodatabaseFeatureTableFuture = geodatabase.createTableAsync(tableDescription);
+
+        // set up the map view to display the feature layer using the loaded [tableFuture] geodatabase feature table
+        geodatabaseFeatureTableFuture.addDoneListener(() -> {
+          try {
+            // get the result of the loaded "LocationHistory" table
+            geodatabaseFeatureTable = geodatabaseFeatureTableFuture.get();
+            // create a feature layer for the map using the GeodatabaseFeatureTable
+            var featureLayer = new FeatureLayer(geodatabaseFeatureTable);
+            mapView.getMap().getOperationalLayers().add(featureLayer);
```

Comment:
we have access to `map` here still, so can use this in favour of the map view instance
```suggestion
            map().getOperationalLayers().add(featureLayer);
```